packages/babel-generator/test/fixtures/typescript/cast-need-parentheses/output.js

```diff
@@ -1,5 +1,21 @@
 (<T> x).y;
 (x as T).y;
 x!.y;
-point1 ?? (point2 as Point);
-((point1 ?? point2) as Point);
\ No newline at end of file
+point1 ?? point2 as Point;
+(point1 ?? point2) as Point;
+a in b as c;
+a in b as c;
+a in (b as c);
+a as b in c;
+a as b in c;
+a | b as c;
+(a | b) as c;
+a | b as c;
+a as b | c;
+(a as b) | c;
+a as (b | c);
```

Comment:
Here parens are not actually necessary, but our AST has a TSParenthesizedType node so they are preserved.